docs/developer/cluster-lifecycle-api-provider-integration.md

```diff
@@ -0,0 +1,60 @@
+# Cluster Lifecycle API provider integration guide
+
+> `THIS IS A WORK IN PROGRESS`
+
+## Context
+> Note: if you haven't yet, reading the [original design doc](https://github.com/aws/eks-anywhere/blob/main/designs/full-cluster-lifecycle-api.md) is a must to execute this integration.
+
+Then main reconciliation logic is ran in the cluster controller. This watches all eks-a CRs and reacts to their changes. However, currently the cluster controller itself only performs some config/setup (controller watches, patch the `Cluster` CR at the end of the reconcile method, set owner refs, etc.) and reconciles deleted `Cluster`s. The main reconciliation logic is delegated to the `ProviderClusterReconciler`'s, one per provider.
+
+`ProviderClusterReconciler` has to handle all the actions needed to reconcile the workload cluster to the state defined in the eks-a CRs. It has complete control over how this is implemented and there are no restrictions imposed in this regard. That said, there are 3 main tasks generic tasks that every `ProviderClusterReconciler` has to perform, in this order:
+1. Reconcile the workload cluster Control Plane to the eks-a CR specifications
+1. Reconcile the workload cluster worker nodes to the eks-a CR specifications
+1. Reconcile the CNI to the eks-a CR specifications
+
+Each provider can add any other additional steps, in order: before, after ir even in between of these 3.
+
+The CNI reconciliation should be executed by using the Cilium reconciler (and other in the future), orchestrated by the main [CNI Reconciler](https://github.com/aws/eks-anywhere/blob/main/pkg/networking/reconciler/reconciler.go). These are designed to work in a reconciliation loop provided a stable Control Plane and and a client to interact with.
+
+## Requirements
+
+* Implement the [`ProviderClusterReconciler` interface](https://github.com/aws/eks-anywhere/blob/main/pkg/controller/clusters/registry.go#L12)
```

Comment:
what is the migration process for providers using the legacy reconciler? Is there a way to "dry-run" this ProviderClusterReconciler during development? How do we decide when to switch over to the new reconciler?

Reply:
What do you mean by "dry-run"?

Once the new implementation is ready for a provider, we will update the CLI to enable it when installing the controller manager deployment.

For development, we have a a feature flag. You can enable it by setting an env var either before management cluster creation or before running tilt.

- [ ] I'll add some details